packages/core/src/lib/Table/Table.styles.ts

```diff
@@ -93,6 +95,9 @@ const TableGrid = styled.div.attrs((props: any) => ({
     & > div.table_header {
         font-weight: 600;
         padding: 12px 0;
+        color: ${(props) => props.headerTextColor ?? 'white'};
+        background-color: ${(props) => props.headerBgColor ?? 'white'} !important;
+        box-shadow: -${(props) => props.headerBgColor ? ((props.columnGapSize + 'px 0px 0px 0px' + props.headerBgColor) ?? '11px'):('0px')}, ${(props) => props.headerBgColor ? ((props.columnGapSize + 'px 0px 0px 0px' + props.headerBgColor) ?? '11px'):('0px')}; // to fill the grip-gap with color
```

Comment:
Looks like you are doing the same conditional checks 2 times, it would be better to create a function for it, will make it simpler for the next dev 😄 
(I like this idea of using box shadow to fill the gap color niceee 🥇 )